src/asset.rs

```diff
@@ -154,6 +156,10 @@ impl Asset {
         hasher.finalize().to_string()
     }
 
+    pub fn extension(&self) -> &str {
```

Comment:
Why?

Reply:
Having `extension` as a public field would allow it to be mutated, I figured this wouldn't make much sense in this context, so I created a getter for it